fallback_builtins.h

```diff
@@ -9,36 +9,37 @@
 #  include "arch/x86/x86_features.h"
 #endif
 
-/* This is not a general purpose replacement for __builtin_ctz. The function expects that value is != 0
- * Because of that assumption trailing_zero is not initialized and the return value of _BitScanForward is not checked
+/* This is not a general purpose replacement for __builtin_ctz. The function expects that value is != 0.
+ * Because of that assumption trailing_zero is not initialized and the return value is not checked.
+ * Tzcnt and bsf give identical results except when input value is 0, therefore this can not be allowed.
+ * If tzcnt instruction is not supported, the cpu will itself execute bsf instead.
+ * Performance tzcnt/bsf is identical on Intel cpu, tzcnt is faster than bsf on AMD cpu.
  */
 static __forceinline int __builtin_ctz(unsigned int value) {
-#ifdef X86_FEATURES
-#  ifndef X86_NOCHECK_TZCNT
-    if (x86_cpu_has_tzcnt)
-#  endif
-        return (int)_tzcnt_u32(value);
-#endif
+    Assert(value != 0);
+# if X86_FEATURES && !(_MSC_VER < 1700)
```

Comment:
```c++
defined(X86_FEATURES)
```
and below.

since the ``X86_FEATURES`` macro is defined without a value:
https://github.com/zlib-ng/zlib-ng/blob/d144fc06bfa575ddfa8143cedd0873f36a2c7568/CMakeLists.txt#L743

Reply:
OMG, should not program before breakfast.. 
Shuffled the code, copy-pasted that in and then completely forgot to add the defined() 🤦‍♂️